packages/python/plotly/plotly/tests/test_core/test_px/test_trendline.py

```diff
@@ -1,14 +1,103 @@
 import plotly.express as px
 import numpy as np
+import pandas as pd
+import pytest
+from datetime import datetime
 
 
-def test_trendline_nan_values():
+@pytest.mark.parametrize("mode", ["ols", "lowess"])
+def test_trendline_results_passthrough(mode):
+    df = px.data.gapminder().query("continent == 'Oceania'")
+    fig = px.scatter(df, x="year", y="pop", color="country", trendline=mode)
+    assert len(fig.data) == 4
+    for trace in fig["data"][0::2]:
+        assert "trendline" not in trace.hovertemplate
+    for trendline in fig["data"][1::2]:
+        assert "trendline" in trendline.hovertemplate
+        if mode == "ols":
+            assert "R<sup>2</sup>" in trendline.hovertemplate
+    results = px.get_trendline_results(fig)
+    if mode == "ols":
+        assert len(results) == 2
+        assert results["country"].values[0] == "Australia"
+        assert results["country"].values[0] == "Australia"
+        au_result = results["px_fit_results"].values[0]
+        assert len(au_result.params) == 2
+    else:
+        assert len(results) == 0
+
+
+@pytest.mark.parametrize("mode", ["ols", "lowess"])
+def test_trendline_enough_values(mode):
+    fig = px.scatter(x=[0, 1], y=[0, 1], trendline=mode)
+    assert len(fig.data) == 2
+    assert len(fig.data[1].x) == 2
+    fig = px.scatter(x=[0], y=[0], trendline=mode)
+    assert len(fig.data) == 2
+    assert fig.data[1].x is None
+    fig = px.scatter(x=[0, 1], y=[0, None], trendline=mode)
+    assert len(fig.data) == 2
+    assert fig.data[1].x is None
```

Comment:
should we raise an error when there are not enough points to compute a trendline?

Reply:
No, because it's easy to get into a situation where you facet on a variable, and then within a given facet there is only one point of a given color, so we don't want to prevent ppl from being able to draw trendlines on all the other colors. This happens e.g. with tips when you facet on day/time and color by sex: there's a day/time/sex with only one data point.

Reply:
makes sense, thanks for the explanation